catroid/src/main/res/values/arrays.xml

```diff
@@ -202,6 +202,61 @@
         <item>@string/brick_set_physics_object_type_fixed</item>
         <item>@string/brick_set_physics_object_type_no</item>
     </string-array>
+    <!--Multilingual activity  -->
 
+    <string-array name="Languages_items" translatable="false">
+
+        <item>Device Language</item>
```

Comment:
~Use `<item>@string/devide_language</item>` and create an entry in the default strings.xml. It'll be translated to other languages by Crowdin.~

~The `translatable` flag is just an info for lint and translators and afaik isn't processed by Android.
All other items don't need to be translated imo.~~

My fault, as @rschluesselbauer suggested, try https://github.com/Catrobat/Catroid/pull/2332#discussion_r125192104 first. I'm still not sure if it will handle single language variations (e.g. English(Australia), English(Canada), English(…))


Creating a good kind of mapping is definitely the way to handle language names ↔ language tags/codes. @rschluesselbauer already has some good ideas here. If you have questions, just ping him for help. He'll definitely help you out.